easy-ui-react/src/Spinner/Spinner.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useId, ReactNode } from "react";
+import { useProgressBar } from "react-aria";
+import { Text } from "../Text";
+import { ThemeColorAliases, IntRange } from "../types";
+import {
+  classNames,
+  getComponentThemeToken,
+  getComponentDesignToken,
+  getComponentToken,
+} from "../utilities/css";
+import styles from "./Spinner.module.scss";
+
+export type ProgressProps = {
+  /**
+   * Mark the `Spinner` as indeterminate when progress is
+   * unknown.
+   */
+  isIndeterminate: true;
+  /**
+   * The current progress
+   */
+  value?: undefined;
+};
+export type IndeterminateProps = {
+  /**
+   * Mark the `Spinner` as indeterminate when progress is
+   * unknown.
+   */
+  isIndeterminate?: false;
+  /**
+   * The current progress
+   */
+  value: IntRange<0, 100>;
+};
+
+export type SpinnerProps = (ProgressProps | IndeterminateProps) & {
+  /**
+   * Size of spinner.
+   * @default md
+   */
+  size?: "sm" | "md" | "xl";
```

Comment:
**question**: should it allow any `size.icon` token?

Reply:
After talking to Rechel, we are going to support all icon sizes